### ISSUE TITLE
doesn't work on raspberry pi?

### ISSUE THREAD

**Issue by brianinmaine** [elisescu/tty-share#3]
```
pi@tiny:~/out $ ./tty-share.lin 
-bash: ./tty-share.lin: cannot execute binary file: Exec format error

```
I didn't try compiling, but it does work great on my other systems. Fun stuff, thank a lot. Just pointing out it doesn't seem to work out of the box for ARM.

**Comment by fallen90** [comment]
@brianinmaine you need to build it from source. 

`
GOARCH=arm64 GOOS=linux go build -o out/tty-share ./tty-share/main.go ./tty-share/pty_master.go
`

**Comment by brianinmaine** [comment]
OK - great! Will do. Seems fun to play with.

**Comment by fallen90** [comment]
@brianinmaine that's great, but weird at the same time, I'm using rpi 3B+ on a raspian buster, and I had to use `GOARCH=arm64` to make it work. anyway as long it works on someone, it works.

**Comment by brianinmaine** [comment]
I have no experience with go, so maybe I was doing something wrong there? I wasn't sure how to set up the environment, so I just did "go get dependency" a few times until it worked. Do you have a reference for the proper way to set up? Also I'm using raspbian buster and used "apt install golang" I think it's 1.11? Probably my fault anyway, but it works so all good. :)

**Comment by mluis** [comment]
it would be great to have documentation on builds for other architectures like mips or arm

**Comment by elisescu** [comment]
Hey! Thanks for trying it out, and sorry for the late reply.

I have added few more details in the Readme file, and also another target in the Makefile to pull the deps manually in order to build it: https://github.com/elisescu/tty-share/commit/16974db44b228eacf819c06e62364ee15fc986ce

 I hope I will have more time to write more documentation. But for now, for building the tty-share locally, on rasperrypi, it should work with the command I wrote here: https://github.com/elisescu/tty-share#building-tty-share-locally.
